### PR TITLE
Makes pie chart component accept data frames

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What this PR does / why we need it**:
To align with other components this changes the pie chart component so that it accepts data frames instead of display values.
There's also some small refactoring around types.

### CODE REVIEW

**Comment by oscarkilhed** [comment]
Just realized some of the new properties on the PieChart component could probably be optional. Gonna have a look.

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx [hunk 1]
```diff
@@ -111,6 +97,15 @@ export const PieChart: FC<Props> = ({
     );
   };
 
+  const fieldDisplayValues = getFieldDisplayValues({
+    fieldConfig,
+    reduceOptions: reduceOptions,
+    data: data,
+    theme: useTheme(),
```
Comment (dprokop) [inline packages/grafana-ui/src/components/PieChart/PieChart.tsx:104]: Hooks should be used as top most calls in a component  (https://ru.react.js.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx [hunk 2]
```diff
@@ -111,6 +97,15 @@ export const PieChart: FC<Props> = ({
...
+    replaceVariables: replaceVariables,
+    timeZone,
+  });
```
Comment (dprokop) [inline packages/grafana-ui/src/components/PieChart/PieChart.tsx:107]: nit:

```suggestion
    fieldConfig,
    reduceOptions,
    data,
    theme,
    replaceVariables,
    timeZone,
  });
```

**Review by dprokop — APPROVED** [review]
Works good. Two nits only :)

**Pull request merged**